src/main/java/com/aws/greengrass/shadowmanager/sync/RequestMerger.java

```diff
@@ -106,10 +107,11 @@ public SyncRequest merge(SyncRequest oldValue, SyncRequest value) {
                     .log("Received bi-directional updates. Converting to a full shadow sync request");
         }
 
-        return returnRequestBasedOnDirection(value, logEvent);
+        return returnRequestBasedOnDirection(value, oldValue, logEvent);
     }
 
-    private BaseSyncRequest returnRequestBasedOnDirection(SyncRequest value, LogEventBuilder logEvent) {
+    private BaseSyncRequest returnRequestBasedOnDirection(SyncRequest value, SyncRequest otherValue,
```

Comment:
not sure i follow?

So the merger merges sync requests for the same shadow, so their related by thing/shadow name.  If merger decides that the two sync requests should be combined into a full sync, then those requests will be added to a list in MergedFullShadowSyncRequest.  I don't think it's a matter of old and current, but more of the reason or history of what individual sync requests were combined to become this full sync.

Reply:
I meant more from the standpoint of code organization/correctness this function is accepting 2 values `SyncRequest` that could theoretically be related to different shadows based on the types but the reality is that one is just a older version of the current SyncRequest. What I am saying is that instead of having as separate variables they should be part of the same object if possible.